src/components/SingleList.jsx

```diff
@@ -1,19 +1,102 @@
-import { useNavigate } from 'react-router-dom';
 import './SingleList.css';
+import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
+import PushPinIcon from '@mui/icons-material/PushPin';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { MaterialListButton, MaterialButton } from './material-buttons';
+import { useAuth, deleteList } from '../api';
+import { ConfirmDialog } from './ConfirmDialog';
+
+export const deletionResponse = {
+	hard: `List deleted permanently.`,
+	soft: `List removed from user view.`,
+};
+
+export function SingleList({
+	item,
+	setListPath,
+	setImportantList,
+	isImportant,
+}) {
+	const [isHovered, setIsHovered] = useState(false);
+	const [open, isOpen] = useState(false);
 
-export function SingleList({ name, path, setListPath }) {
 	const navigate = useNavigate();
+	const { user } = useAuth();
+	const userId = user?.uid;
+	const userEmail = user?.email;
 
-	function handleClick() {
+	const { name, path } = item;
+	const listPath = path.slice(0, path.indexOf('/'));
+
+	const handleNavigate = () => {
 		setListPath(path);
 		setTimeout(() => {
 			navigate('/list');
 		}, 200);
-	}
+	};
+
+	const handleImportantList = () => {
+		setImportantList(path);
+	};
+
+	const toggleDialog = () => {
+		isOpen((prev) => !prev);
+	};
+
+	const handleDelete = async () => {
+		const deletionType = listPath === userId ? 'hard' : 'soft';
+
+		try {
+			await deleteList(deletionType, path, userId, userEmail);
+			alert(deletionResponse[deletionType]);
+		} catch (error) {
+			alert(
+				`List was not deleted. Error: ${error.message ? error.message : error}`,
+			);
+		}
+
+		return;
+	};
+
+	const pinStyle = {
+		visibility: isImportant || isHovered ? 'visible' : 'hidden',
```

Comment:
Maybe a Danielle problem, but I had to read and re-read this a few times. I wonder if there's a way to make this more readable like adding a comment or something?